kombu/connection.py

```diff
@@ -1039,6 +1039,9 @@ def prepare(self, resource):
         resource._debug('acquired')
         return resource
 
+    def acquire(self, block: bool = False, timeout: int | float | None = None) -> Connection:
```

Comment:
is this method newly introduced?

Reply:
Not new, ConnectionPool inherits from Resource 

Resource has an acquire method, but since Resource is an abstract class (abstracting the connection as a resource), this `-> Connection` type annotation should not be added to the acquire method of Resource, so I added an acquire method to ConnectionPool and added the  I added an acquire method to the ConnectionPool and added the `-> Connection` type annotation to the ConnectionPool to make it semantic

`kombu/connection.py`

```python
class ConnectionPool(Resource):
    """Pool of connections."""
```

`kombu/resource.py`

```python
class Resource:
    """Pool of resources."""

    def acquire(self, block=False, timeout=None):
        """Acquire resource.

        Arguments:
            block (bool): If the limit is exceeded,
                then block until there is an available item.
            timeout (float): Timeout to wait
                if ``block`` is true.  Default is :const:`None` (forever).

        Raises:
            LimitExceeded: if block is false and the limit has been exceeded.
```